enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,314 @@
+for updating a CA certificate in each route object when the certificate is expired, compromised, or for
+any other reason in need of replacement.
+
+### User Stories
+- As an OpenShift cluster administrator, I want to configure a custom CA certificate bundle which OpenShift
+  router should use for verifying services' certificates, instead of having to configure the same as
+  destination CA Certificate in each of the created route objects.
+- As an OpenShift developer, I want to create a CA trust bundle for OpenShift Router usage, which includes
+  the OpenShift service CA and the admin provided CA bundle as the default CA for verifying the service's
+  certificate when a destination CA certificate is not configured for a route.
+- As an OpenShift cluster administrator, I want OpenShift router to update its configuration whenever the
+  OpenShift service CA certificate or the admin-configured CA certificate bundle is updated or deleted.
+- As an OpenShift cluster administrator, I want OpenShift router to verify services' certificates with just
+  the configured destination CA Certificate and not the default destination CA certificate.
+
+### Goals
+
+- An OpenShift administrator can configure a CA certificate bundle and expect OpenShift router to use it
+  for verifying a service's certificate when destination CA Certificate isn't specified in the route object
+  created for the service.
+- Openshift router should seamlessly reload the OpenShift service CA certificate or the
+  administrator-configured CA certificate bundle when either is modified.
+
+### Non-Goals
+
+- OpenShift administrator defined CA certificate bundle will be used only for the traffic through OpenShift
+  router and for the routes created with re-encrypt termination type, any other scenario is outside the
+  scope of this proposal.
+- The common CA bundle created by the operator is for all ingress controllers usage, and defining a CA
+  bundle specific for each IngressController usage is out of scope.
+
+## Proposal
+
+Change OpenShift Ingress Operator to:
+1. Create a CA trust bundle which will contain the OpenShift service CA certificate and the custom
+   user-supplied CA certificates.
+2. Make the new common certificate available to the OpenShift Router as the default destination CA bundle
+   file to be used for verifying the services' certificate.
+3. Updates the CA bundle file whenever either of the certificates change.
+
+Update router to:
+1. Watch the common CA bundle and update its configuration internally when needed.
+
+### Workflow Description
+
+- An OpenShift administrator configures the list of trusted CA certificates as a ConfigMap, named
+  specifically `admin-ca-bundle` in `openshift-config` namespace.
+- OpenShift Ingress Operator creates `ingress-ca-bundle` ConfigMap in `openshift-ingress` namespace by
+  concatenating the admin configured CA certificates and the OpenShift service CA certificate stored in
```

Comment:
This looks improper.  If a cluster-admin is saying, "verify using this CA bundle", then it should use "this CA bundle".  not "this CA bundle and the other ones I have".  In other kube binaries and flags, specifying a CA bundle means, "use exactly this one".

Reply:
Here we are combining the openshift's own CA certificate and the user provided ca bundle as common ca trust to verify the service's certificate behind a route, which is used like system ca-trust bundle. Please correct me.